NEWS.md

```diff
@@ -4,14 +4,75 @@ User-visible changes worth mentioning.
 
 ---
 
-## 2.2.2
-
-- [#654] Fixed `check_requirements` not to abort when ActiveRecord
-    `oauth_applications` table does not have the `scopes` column.
-    (Issue #653)
-- [#599] Included MassAssignmentSecurity in mixins if
-    ::ProtectedAttributes is defined.
-- [#650] Only fallback i18n when it is not set.
+- [#778] Use the remaining time that a token is still valid when building the
+  redirect URI for the implicit grant flow
+
+## 3.1.0
+
+- [#736] Existing valid tokens are now reused in client_credentials flow
+- [#749] Allow user to raise authorization error with custom messages.
+  Under `resource_owner_authenticator` block a user can
+  `raise Doorkeeper::Errors::DoorkeeperError.new('custom_message')`
+- [#762] Check doesn’t abort the actual migration, so it runs
+- [#722] `doorkeeper_forbidden_render_options` now supports returning a 404 by
+  specifying `respond_not_found_when_forbidden: true` in the
+  `doorkeeper_forbidden_render_options` method.
+- [#734] Simplify and remove duplication in request strategy classes
+
+## 3.0.1
+
+- [#712] Wrap exchange of grant token for access token and access token refresh
+  in transactions
+- [#704] Allow applications scopes to be mass assigned
+- [#707] Fixed order of Mixin inclusion and table_name configuration in models
+- [#712] Wrap access token and refresh grants in transactions
+- Adds JRuby support
+- Specs, views and documentation adjustments
+
+## 3.0.0
+
+### Other changes
+
+- [#693] Updates `en.yml`.
+
+## 3.0.0 (rc2)
+
+### Backward incompatible changes
+
+- [#678] Change application-specific scopes to take precedence over server-wide
+  scopes. This removes the previous behavior where the intersection between
+  application and server scopes was used.
+
+### Other changes
+
+- [#671] Fixes `NoMethodError - undefined method 'getlocal'` when calling
+  the /oauth/token path. Switch from using a DateTime object to update
+  AR to using a Time object. (Issue #668)
+- [#677] Support editing application-specific scopes via the standard forms
+- [#682] Pass error hash to Grape `error!`
+- [#683] Generate application secret/UID if fields are blank strings
+
+## 3.0.0 (rc1)
+
+### Backward incompatible changes
+
+- [#648] Extracts mongodb ORMs to
+  https://github.com/doorkeeper-gem/doorkeeper-mongodb. If you use ActiveRecord
+  you don’t need to do any change, otherwise you will need to install the new
+  plugin.
+- [#665] `doorkeeper_unauthorized_render_options(error:)` and
+  `doorkeeper_forbidden_render_options(error:)` now accept `error` keyword
+  argument.
+
+### Removed deprecations
+
+- Removes `doorkeeper_for` deprecation notice.
+- Remove `applications.scopes` upgrade notice.
+
+
+## 2.2.2 (unreleased)
+
+>>>>>>> 56bfa579484ae9e554028b081227ec8d1b74c21d
```

Comment:
Just get's our branch up to date.

Best,

Ross

マネーツリー株式会社

TEL：090-9954-1846
〒150-0001 東京都渋谷区神宮前 3-27-15



On Tue, Feb 16, 2016 at 2:34 PM, Mark <notifications@github.com> wrote:

> In NEWS.md
> <https://github.com/moneytree/doorkeeper/pull/14#discussion_r52970367>:
>
> > +  https://github.com/doorkeeper-gem/doorkeeper-mongodb. If you use ActiveRecord
> > +  you don’t need to do any change, otherwise you will need to install the new
> > +  plugin.
> > +- [#665] `doorkeeper_unauthorized_render_options(error:)` and
> > +  `doorkeeper_forbidden_render_options(error:)` now accept `error` keyword
> > +  argument.
> > +
> > +### Removed deprecations
> > +
> > +- Removes `doorkeeper_for` deprecation notice.
> > +- Remove `applications.scopes` upgrade notice.
> > +
> > +
> > +## 2.2.2 (unreleased)
> > +
> > +>>>>>>> 56bfa579484ae9e554028b081227ec8d1b74c21d
>
> @rsharrott <https://github.com/rsharrott> does this do anything else for
> us other than get us up to date? e.g. fix those 500s, etc?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/moneytree/doorkeeper/pull/14/files#r52970367>.
>